src/App/mermaidData/syncApiDataIntoOfflineStorage/useInitializeSyncApiDataIntoOfflineStorage.js

```diff
@@ -41,14 +41,16 @@ export const useInitializeSyncApiDataIntoOfflineStorage = ({
     const isProjectsListPage =
       location.pathname === '/projects' || location.pathname === '/projects/'
 
+    const isUserDeniedProjectAccessPage = location.pathname.includes('noProjectAccess')
+
     const isProjectsListPageAndOnline = isProjectsListPage && isOnlineAndReady
 
     const isInitialLoadOnProjectPageAndOnline =
       isPageReload.current && isProjectPage && isOnlineAndReady
     const isNotInitialLoadOnProjectPageAndOnline =
       !isPageReload.current && isProjectPage && isOnlineAndReady
 
-    if (isOfflineAndReadyAndAlreadyInitiated) {
+    if (isOfflineAndReadyAndAlreadyInitiated || isUserDeniedProjectAccessPage) {
       setIsOfflineStorageHydrated(true)
```

Comment:
we need this to make sure the loader clears and we can see the new page just made